merino/providers/base.py

```diff
@@ -33,11 +44,11 @@ async def initialize(self) -> None:
         ...
 
     @abstractmethod
-    async def query(self, query: str) -> list[BaseSuggestion]:
+    async def query(self, srequest: SuggestionRequest) -> list[BaseSuggestion]:
         """Query against this provider.
 
         Args:
-          - `query`: the query string.
+          - `SuggestionRequest`: the query requested.
```

Comment:
```suggestion
          - `SuggestionRequest`: the suggestion request object.
```

Since `geolocation` is also part of `SuggestionRequest`.